### DISCUSSION TITLE
How to call an method (invoke) with args Vector3 or System.Int64?

### DISCUSSION THREAD

**Discussion by EduModsS** [vfsfitvnm/frida-il2cpp-bridge#382] [Q&A]
class FoodChamber : UnityEngine.MonoBehaviour {
    System.Void wProcessFood(UnityEngine.Vector3 pos); // 0x012f50a4
}

I know how to invoke the method, however, how to invoke it, with some modified vector3 in place of pos? like : myMethod.invoke(vector3(x,y,z));

and how to edit / get value in Int64 (original value) in this type of field?

lUnityEngine.Vector3 distanciaCamPlayer; // 0x2c,  how to set value in vector3? I think it would be similar to the method but with .value = vector(x,y,z);

class CharGrowthMetaData : GameMetaData {
    System.Int32 atk; // 0x1c
    System.Int64 hp; // 0x1c
}

how to get the original value in Int64 / Int32 and set the value in Int64 / In32 ?

please give me some examples

**Comment by VAROME** [comment]
show me your codes maybe i can help .

**Reply by EduModsS** [reply]
I no longer have the codes saved, I just wanted an example of use for me to test

**Comment by bluewave41** [comment]
Either get a vector from memory using [Il2Cpp.gc.choose](https://github.com/vfsfitvnm/frida-il2cpp-bridge/wiki/Snippets#heap-scanning) and edit that

Or create your own. Dumping with `Il2Cpp.dump()` shows you that `struct UnityEngine.Vector3` belongs to` UnityEngine.CoreModule`

```
const unityCore = Il2Cpp.domain.assembly('UnityEngine.CoreModule').image;
const vector = unityCore.class('UnityEngine.Vector3').alloc();
vector.method('.ctor').invoke(x, y, z);
```

and pass into the function.

To get the values from that vector you would do

`vector.field('x').value;`

To get the value and

`vector.field('x').value = 2`

to change the value.
